java/cuvs-java/src/test/java/com/nvidia/cuvs/TieredIndexIT.java

```diff
@@ -0,0 +1,254 @@
+              .withIndexParams(indexParams)
+              .build();
+
+      CagraSearchParams searchParams =
+          new CagraSearchParams.Builder(resources).withMaxIterations(20).build();
+
+      TieredIndexQuery query =
+          new TieredIndexQuery.Builder()
+              .withTopK(3)
+              .withQueryVectors(queries)
+              .withSearchParams(searchParams)
+              .build();
+
+      log.info("Searching initial index...");
+      SearchResults initialResults = index.search(query);
+      log.info("Initial search results: {}", initialResults.getResults());
+      assertEquals(expectedInitialResults, roundResults(initialResults.getResults()));
+
+      log.info("Extending index...");
+      index.extend().withDataset(extensionVectors).execute();
+
+      log.info("Searching extended index...");
+      SearchResults extendedResults = index.search(query);
+      log.info("Extended search results: {}", extendedResults.getResults());
+      assertEquals(expectedExtendedResults, roundResults(extendedResults.getResults()));
+    }
+  }
+
+  /**
+   * Tests error handling and parameter validation.
+   */
+  @Test(expected = IllegalArgumentException.class)
+  public void testErrorHandling() throws Throwable {
+    try (CuVSResources resources = CuVSResources.create()) {
+      CagraIndexParams cagraParams =
+          new CagraIndexParams.Builder().withGraphDegree(4).withIntermediateGraphDegree(8).build();
+
+      TieredIndexParams indexParams =
+          new TieredIndexParams.Builder().minAnnRows(2).withCagraParams(cagraParams).build();
+
+      TieredIndex.newBuilder(resources)
+          .withIndexParams(indexParams)
+          .withDataset((float[][]) null)
+          .build();
+    }
+  }
+
+  /**
+   * Tests search with different K values.
+   */
+  @Test
+  public void testDifferentKValues() throws Throwable {
+    float[][] dataset = {
+      {0.0f, 0.0f},
+      {1.0f, 1.0f},
+      {2.0f, 2.0f},
+      {3.0f, 3.0f},
+      {4.0f, 4.0f}
+    };
+
+    float[][] queries = {{0.1f, 0.1f}};
+
+    try (CuVSResources resources = CuVSResources.create()) {
+      CagraIndexParams cagraParams =
+          new CagraIndexParams.Builder().withGraphDegree(4).withIntermediateGraphDegree(8).build();
+
+      TieredIndexParams indexParams =
+          new TieredIndexParams.Builder().minAnnRows(2).withCagraParams(cagraParams).build();
+
+      TieredIndex index =
+          TieredIndex.newBuilder(resources)
+              .withDataset(dataset)
+              .withIndexParams(indexParams)
+              .build();
+
+      TieredIndexQuery query1 =
+          new TieredIndexQuery.Builder()
+              .withTopK(1)
+              .withQueryVectors(queries)
+              .withSearchParams(
+                  new CagraSearchParams.Builder(resources).withMaxIterations(20).build())
+              .build();
+
+      SearchResults results1 = index.search(query1);
+      assertEquals(1, results1.getResults().get(0).size());
```

Comment:
```suggestion
      assertEquals(1, results1.getResults().getFirst().size());
```
Is there a reason we aren't checking the result values, instead of the right number of results?